### PR TITLE
fix: improve errors for unsupported methods called for a non custom-searcher [MLG-28]

### PR DESCRIPTION
**Pull request by nrajanee**
## Description

fix: improve errors for unsupported methods called for a non custom-searcher 

## Test Plan

N/A

### CODE REVIEW

**Comment by nrajanee** [comment]
Maybe some of this could've been avoided if it was a custom searcher (one level up) instead of a custom-search method. Will look into changing it to custom searcher after the first release.

File: master/pkg/searcher/searcher.go [hunk 1]
```diff
@@ -59,6 +59,10 @@ func NewSearcher(seed uint32, method SearchMethod, hparams expconf.Hyperparamete
 	}
 }
 
+func nonCustomSearchMethodError(method SearchMethod, unsupportedOp string) error {
+	return fmt.Errorf("%s is not supported for search method of type %T", unsupportedOp, method)
+}
```
[the remark this answers is not in the document]
Comment (nrajanee) [reply master/pkg/searcher/searcher.go:64]: done! thank you.

[the remark this answers is not in the document]
Comment (nrajanee) [reply master/pkg/searcher/searcher.go:64]: done!

File: master/internal/experiment.go [hunk 2]
```diff
@@ -599,7 +599,11 @@ func (e *experiment) processOperations(
 			e.TrialSearcherState[op.RequestID] = state
 			updatedTrials[op.RequestID] = true
 		case searcher.SearcherProgress:
-			e.searcher.SetCustomSearcherProgress(op.Progress)
+			err := e.searcher.SetCustomSearcherProgress(op.Progress)
+			if err != nil {
+				ctx.Respond(status.Error(codes.Internal, err.Error()))
+			}
+
```
Comment (dzhu) [inline master/internal/experiment.go:606]:
```suggestion
			if err := e.searcher.SetCustomSearcherProgress(op.Progress); err != nil {
				ctx.Respond(status.Error(codes.Internal, err.Error()))
			}
```

**Review by dzhu — APPROVED** [review]

**Pull request merged**